eos/utils/expression-visitors.cc

```diff
@@ -688,8 +689,37 @@ namespace eos::exp
     Expression
     ExpressionCacher::operator() (const ObservableExpression & e)
     {
+        const auto & kinematics_aliases = e.kinematics_specification.aliases;
+
+        for (const auto & alias : kinematics_aliases)
+        {
+            e.observable->kinematics().alias(alias.first, alias.second);
+        }
+
+        // find out which kinematic variables the observable is supposed to use
+        std::set<std::string> kinematic_set;
+        const auto &          entries = impl::observable_entries;
+        const auto &          it      = entries.find(e.observable->name());
+        if (it != entries.end())
+        {
+            const auto entry = it->second;
+            kinematic_set.insert(entry->begin_kinematic_variables(), entry->end_kinematic_variables());
+        }
+
+        // add the ids of the used kinematic variables to kinematic_user
+        for (const auto & kinematic_name : kinematic_set)
+        {
+            e.observable->uses_kinematic(e.observable->kinematics()[kinematic_name].id());
+        }
+
```

Comment:
This piece of code is in the wrong place. After your previous commit, all observables must track their own use of kinematic variables. This type of code needs to go into ``ConcreteObservable<T_>``, amongst other things.

Maybe we should have a brief meeting about this? Might be easier and faster?